tools/gulp/docs/index.js

```diff
@@ -183,47 +178,56 @@ module.exports = (gulp, shared) => {
    * happens within a chain of promises.
    * @return {Promise}
    */
-  gulp.task('docs:generate-pages', () => {
+  gulp.task('docs:generate-pages', async function() {
     dutil.logMessage(
       '📝 ',
       'Creating HTML pages from Sass comments and Markdown pages'
     );
 
     const packages = shared.packages.map(pkg => `packages/${pkg}/src/`);
     const mask = /^(?!.*\.(example|test)).*\.(css|less|sass|scss|jsx)$/;
-    return kss
+
+    // Parse Markdown files, and return the data in the same format as a KssSection
+    const markdownPagesData = await convertMarkdownPages(
+      shared.rootPath,
+      shared.packages
+    );
+
+    /**
+     * Parse CSS and JS comments, and form an array of KssSection objects:
+     * kss-node.github.io/kss-node/api/master/module-kss.KssSection.html
+     * @return {Array} KssSections
+     */
+    const kssSections = await kss
       .traverse(packages, { mask })
       .then(styleguide =>
-        /**
-         * Parse CSS and JS comments, forming the initial array of pages.
-         * CSS and JS comments are parsed and an array of KSS Section objects is
-         * generated: kss-node.github.io/kss-node/api/master/module-kss.KssSection.html
-         * @return {Array} KssSections
-         */
         Promise.all(
           styleguide.sections().map(kssSection =>
             // Cleanup and extend the section's properties
             processKssSection(kssSection, shared.rootPath)
           )
         )
-      )
-      .then(kssSections =>
```

Comment:
ℹ️ Taking advantage of `async`/`await` support and broke this chain into something easier to read.